[Bugfix] Fix accessibility for the CallingView

vhuseinova-msft opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* Fix issue that video view is not available for VoiceOver when removed participants joined
* Improved info header accessibility behaviour

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```


## What to Check
Verify that the following are valid
* Info header is availble for VoiceOver
* Remote participants names are available for VoiceOver

vhuseinova-msft committed:
Fixed video view and info header availability for VoiceOver on the Calling screen

jimchou-dev commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewModel.swift:
```diff
@@ -155,6 +155,6 @@ class CallingViewModel: ObservableObject {
         let isLocalUserInfoNotEmpty = isLocalCameraOn || !displayName.isEmpty
         isVideoGridViewAccessibilityAvailable = !isLobbyOverlayDisplayed &&
-        !isParticipantGridDisplayed && isLocalUserInfoNotEmpty
+        ((!isParticipantGridDisplayed && isLocalUserInfoNotEmpty) || isParticipantGridDisplayed)
```
Can this logic just be simplified to the following?
```
 isVideoGridViewAccessibilityAvailable = !isLobbyOverlayDisplayed &&
        (isLocalUserInfoNotEmpty || isParticipantGridDisplayed)
```

JoshuaLai commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderViewModel.swift:
```diff
@@ -18,9 +18,7 @@ class InfoHeaderViewModel: ObservableObject {
-    private var shouldDisplayInfoHeader: Bool {
-        callingStatus != .inLobby
-    }
+    private var shouldDisplayInfoHeader: Bool!
```
Why are we `!` on this var? Recommend we default to `false`

vhuseinova-msft replied:
I remember there was a comment from @AmyL219 on 1 of other PR to use this approach as the property will be initialized in init (in a function call)

JoshuaLai replied:
But this isn't very future proof. If there is a refactor in the init function by a new developer, they might not be familiar with this and it will cause potential issues.

vhuseinova-msft replied:
yes, you're right. Do we have any team agreement regarding this?

vhuseinova-msft replied:
Updated to have a default value for now

kevinyulu reviewed (dismissed):
tested in iPhone XR, group and teams call.  info header banner stays when voice over is on. banner is resumed to be hidden after 3s when voice over is turned off. Accessibility works in banner and participant list drawer. lgtm

vhuseinova-msft committed:
Code review updates.
Fixed issue when isVideoGridViewAccessibilityAvailable wasn't updated with new values for isLobbyOverlayDisplayed and isParticipantGridDisplayed
